src/vm.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'config'
+require 'libvirt'
+require 'metal_log'
+
+module Metalware
+  class Vm
+    def initialize(libvirt_host, node, *args)
+      @libvirt ||= Libvirt.open("qemu://#{libvirt_host}/system")
+      @node = node
+      @args = args
+    end
+
+    def info
+      { state: state }
+    end
+
+    def run(cmd)
+      case cmd
+      when 'kill'
+        puts "Killing node #{@node}.."
+        domain.destroy
+      when 'on'
+        puts "Powering up node #{@node}.."
+        domain.create
+      when 'off'
+        puts "Powering down node #{@node}.."
+        domain.shutdown
+      when 'reboot'
+        puts "Rebooting node #{@node}.."
+        domain.reboot
+      when 'status'
+        puts "#{@node}: Power state: #{info[:state]}"
+      else
+        raise MetalwareError, "Invalid command: #{cmd}"
+      end
+    end
+
+    def console
+      puts "Attempting to connect to node #{@node}"
+      domain.open_console if running?
+    end
+
+    # Creates a new VM
+    # - storage: Rendered Libvirt storage XML, used to create the root disk
+    # - vm: Rendered Libvirt domain XML, used to create the domain
+    def create(disk_tpl, domain_tpl)
```

Comment:
Does this method need to have these arguments requiring that the correct templates for the current node be passed in in order to create it? If possible, it would be clearer to me to have this method take no arguments and have this class render the correct templates for the current node, rather than requiring anyone using this class need to know what these arguments should be.

Reply:
Yep, that makes sense - will move this inside of `Vm`!